src/libraries/ArrayUtil.sol

```diff
@@ -366,24 +460,40 @@ library ArrayUtil {
         }
     }
 
-    function fill(int256[] memory x, int256 v) internal pure returns (int256[] memory) {
+    /*
+    @dev this function modifies memory x IN PLACE. Fills x with value v
+    */
+    function fill(int256[] memory x, int256 v) internal pure {
         for (uint256 i = 0; i < x.length; i++) {
             x[i] = v;
         }
-        return x;
     }
 
+    /*
+    @dev modifies memory a IN PLACE. Populates a starting at index z with values from b.
+    */
     function populate(
         uint256[] memory a,
         uint256[] memory b,
         uint256 z
-    ) internal pure returns (uint256[] memory) {
+    ) internal pure {
+        for (uint256 i = 0; i < a.length; i++) {
+            a[z + i] = b[i];
+        }
+    }
+
+    function populate(
+        int256[] memory a,
+        int256[] memory b,
+        uint256 z
+    ) internal pure returns (int256[] memory) {
```

Comment:
@hashnoteleo should keep this version the same as the one above (in regards to return values)

Reply:
made this change